### PR TITLE
feat(server): add Vercel adapter@kitten/server/vercel adapter

### PR DESCRIPTION
**Pull request by kitten**
# Why

- Add adapter and guide for publishing Expo Router v3 websites/api routes to Vercel.

# How

- Add an adapter based on [Remix's one](https://github.com/remix-run/remix/blob/6a104032e22188c906dcb7626c4eb3169de778c9/packages/remix-vercel/server.ts) and along the lines of what the one for Express is doing
- Imports of types for `VercelRequest` and `VercelResponse` are omitted and replaced by ones from Node's `http`, since the former only extend these and we don't need the whole `@vercel/node` package

# Test Plan

As described in the steps in the readme, we can set up a quick Expo Web app with an API route, and run `expo export -p web`. This has a couple of gotchas:
- For Vercel's v3 format of the `vercel.json` the build output must be set to `dist/` for it to find the public assets (at which point the filesystem feature picks them up automatically)
- For Vercel's v2 format the `vercel-build` script must be populated in the `package.json` for the `@vercel/static-build` runtime to pick up our public assets, separately from the function
- Without `includeFiles` (placement differs between v2 & v3) the `dist/` folder won't be accessible to the server function
- In Vercel's v3 format, the `api/*` folder with the middleware adapter must be placed next to the `vercel.json`. If it's not there, many paths will have to be changed (untested). That's why the instructions currently place it at `api/index.js`

### CODE REVIEW

**Review by EvanBacon — APPROVED** [review]
lgtm

File: docs/pages/preview/api-routes.mdx [hunk 1]
```diff
@@ -364,6 +365,137 @@ If you're using any environment variables or **.env** files, add them to Netlify
...
+  /* @end */
+  mode: process.env.NODE_ENV,
+});
+```
+
+</Step>
+
+<Step label="2">
+
+Create a Vercel configuration file at the root of your project to redirect all requests to the server function.
```
Comment (amandeepmittal) [inline docs/pages/preview/api-routes.mdx:391]:
```suggestion
Create a Vercel configuration file (**vercel.json**) at the root of your project to redirect all requests to the server function.
```

Let's also add the file name here.

File: docs/pages/preview/api-routes.mdx [hunk 2]
```diff
@@ -364,6 +365,137 @@ If you're using any environment variables or **.env** files, add them to Netlify
...
+    },
+    {
+      "src": "/(.*)",
+      "dest": "/api/index.js"
+    }
+  ]
+}
+```
+
+The **legacy** version of the `vercel.json` needs a `@vercel/static-build` runtime to serve your assets from the `dist/` output folder.
```
Comment (amandeepmittal) [inline docs/pages/preview/api-routes.mdx:433]:
```suggestion
The **legacy** version of the **vercel.json** needs a `@vercel/static-build` runtime to serve your assets from the **dist** output folder.
```

To maintain consistency in our docs, we prefer bold file/directory names as [per our writing styleguide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md#filenames-as-bold-text).

File: docs/pages/preview/api-routes.mdx [hunk 3]
```diff
@@ -364,6 +365,137 @@ If you're using any environment variables or **.env** files, add them to Netlify
...
+
+</Tab>
+<Tab label="vercel.json v3">
+
+> **warning** When used as is, this `vercel.json` configuration makes server code in `/dist/_expo/functions/*` publicly accessible.
```
Comment (amandeepmittal) [inline docs/pages/preview/api-routes.mdx:438]:
```suggestion
> **warning** When used as is, this **vercel.json** configuration makes server code in **dist/_expo/functions/&#42;** publicly accessible.
```

File: docs/pages/preview/api-routes.mdx [hunk 4]
```diff
@@ -364,6 +365,137 @@ If you're using any environment variables or **.env** files, add them to Netlify
...
+    {
+      "source": "/(.*)",
+      "destination": "/api/index.js"
+    }
+  ]
+}
+```
+
+The newer version of the `vercel.json` does not use `routes` and `builds` configuration options anymore,
+and serves your public assets from the `dist/` output folder automatically.
```
Comment (amandeepmittal) [inline docs/pages/preview/api-routes.mdx:467]:
```suggestion
The newer version of the **vercel.json** does not use `routes` and `builds` configuration options anymore and serves your public assets from the **dist** output folder automatically.
```

**Review by amandeepmittal — APPROVED** [review]
Lgtm 👍 

Left some suggestions for minor formatting/styling issues.

**Pull request merged**